src/solver/application.cpp

```diff
@@ -120,6 +120,38 @@ void checkAdqPatchContainsAdqPatchArea(const bool adqPatchOn, const Antares::Dat
             throw Error::NoAreaInsideAdqPatchMode();
     }
 }
+// Number of columns for Fuel & CO2 cost in thermal clusters must be one, or same as the number of
+// TS
+void checkFuelAndCo2ColumnNumber(const Antares::Data::AreaList& areas)
+{
+    bool error = false;
+    for (uint i = 0; i < areas.size(); ++i)
+    {
+        const auto& area = *(areas.byIndex[i]);
+        for (uint j = 0; j != area.thermal.clusterCount(); ++j)
+        {
+            const auto& cluster = *(area.thermal.clusters[j]);
+            bool setCostManual = cluster.costgeneration == Antares::Data::setManually;
```

Comment:
```suggestion
            bool setCostManual = cluster.costgeneration == Antares::Data::setManually;
            if (setCostManual)
                return;
```
So you can remove `!setCostManual` from the 2 next conditions.